packages/extension-dapp/README.md

```diff
@@ -35,4 +35,9 @@ api.setSigner(injector.signer);
 api.tx.balances
   .transfer('5C5555yEXUcmEJ5kkcCMvdZjUo7NGJiQJMS7vZXEeoMhj3VQ', 123456)
   .signAndSend('5DTestUPts3kjeXSTMyerHihn1uwMfLj8vU8sqF7qYrFabHE', (status) => { ... });
+
+// retrieve a RPC provider from a particular source
+// if the extension does not expose this provider, it will return an error.
+const provider = web3RpcProvider('polkadot-js', { type: 'WsProvider', payload: 'http://locahost:9944' });
```

Comment:
The idea is that when the injected page already contains a pre-configured provider as set. So effectively they can, without doing anything else, just use the providers.

So the interface should just provide a list of all providers, all of them already configured as per the specific extensions. So some additional metadata can be provided -

- source
- name of provider source, e.g. W3F or Parity or ...
- network, e.g Kusama or Custom...
- type (full or light)